nntrainer/tensor/manager.cpp

```diff
@@ -2,27 +2,138 @@
 /**
  * Copyright (C) 2020 Parichay Kapoor <pk.kapoor@samsung.com>
  *
- * @file	manager.cpp
- * @date	2 Dec 2020
- * @brief	This is NNtrainer manager for all weights, i/o and intermediate
+ * @file   manager.cpp
+ * @date   2 Dec 2020
+ * @brief  This is NNtrainer manager for all weights, i/o and intermediate
  * tensors
- * @see		https://github.com/nnstreamer/nntrainer
- * @author	Parichay Kapoor <pk.kapoor@samsung.com>
- * @bug		No known bugs except for NYI items
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Parichay Kapoor <pk.kapoor@samsung.com>
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug    No known bugs except for NYI items
  *
  */
 
+#ifdef __ANDROID__
+#include <android/sharedmem.h>
+#endif
+
+#ifdef DEBUG
+#include <cassert>
+#endif
+
+#include <fcntl.h>
 #include <functional>
+#include <limits>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
 #include <vector>
 
 #include <manager.h>
+#include <nntrainer_log.h>
 
 namespace nntrainer {
+MMapedMemory::MMapedMemory(size_t size, bool allocate_fd) :
+  fd(-1),
+  buf(nullptr),
+  buf_size(0) {
+
+#ifndef __ANDROID__
+  if (allocate_fd) {
+    /// @todo create a file in tmpfs and bind to memfs
+    /// memfd_create is not available for number of platforms so this is
+    /// commented
+    // auto fd_ = memfd_create("", 0);
+    // if (fd_ < 0) {
+    //   throw std::runtime_error("[Manager] creating mem fd failed");
+    // }
+    // if (ftruncate(fd_, size) < 0) {
+    //   throw std::runtime_error("[Manager] truncating fd failed");
+    // }
+    ml_logi("[MMapedMemory] fd creation is not supported in this platform");
+    allocate_fd = false;
+  }
+#endif
+  int fd_ = -1;
+  void *buf_ = nullptr;
```

Comment:
looks duplicate with line 37

Reply:
This is intended for the exception safety guarantee.
`fd` and `buf` aren't assigned if the construction fails.